parallel.go

```diff
@@ -283,7 +290,40 @@ func (r Parallel) SearchValue(ctx context.Context, key string, opts ...ropts.Opt
 	resCh, err := r.forKey(key).search(ctx, func(ri routing.IpfsRouting) (<-chan []byte, error) {
```

Comment:
We should return if `err` is non-nil. I believe `resCh` can be nil in that case (and block forever on reads).